app/src/main/java/org/sopt/dosopttemplate/SignUpActivity.kt

```diff
@@ -0,0 +1,57 @@
+package org.sopt.dosopttemplate
+
+import android.content.Intent
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import com.google.android.material.snackbar.Snackbar
+import org.sopt.dosopttemplate.databinding.ActivitySignUpBinding
+
+class SignUpActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivitySignUpBinding
+    private lateinit var startLoginActivityForResult: ActivityResultLauncher<Intent>
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySignUpBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        initSignBtnClickListener()
+
+        startLoginActivityForResult = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+            if (result.resultCode == RESULT_OK) {
+                val intent = result.data
+                if (intent != null) {
+                    val loginSuccessful = intent.getBooleanExtra("loginSuccessful", false)
+                    if (loginSuccessful) {
+                        finish()
+                    }
+                }
+            }
+        }
+    }
+
+    fun initSignBtnClickListener() {
+        binding.signButton.setOnClickListener {
+            if (binding.idEdit.text.isNullOrBlank() || binding.pwEdit.text.isNullOrBlank() || binding.nameEdit.text.isNullOrBlank() || binding.mbtiEdit.text.isNullOrBlank()) {
+                makeSnackbar("모든 정보를 입력해 주세요.")
+            } else if (binding.idEdit.text.toString().length in 6..10 && binding.pwEdit.text.toString().length >= 8 && binding.pwEdit.text.toString().length <= 12 && binding.nameEdit.text.toString().trim().isNotEmpty() && binding.mbtiEdit.text.toString().length == 4) {
```

Comment:
```suggestion
            } else if (binding.idEdit.text.toString().length in 6..10 && binding.pwEdit.text.toString().length in 8..12 && binding.nameEdit.text.toString().trim().isNotEmpty() && binding.mbtiEdit.text.toString().length == 4) {
```

비밀번호도 .. 으로 작성하는 게 좋을 것 같습니다 ~